### PR TITLE
Putting routing component

### LINKED ISSUE #316: Put Routing components
I am already working on it, but this drags out some bigger issues, so I wanted to write them down. 

- [x] Save api key somewhere so all components can share it : Currently, search component is the only component which requires api  key, so users is passing the key when initializing search component. I think it will make sense to pass the api key with initialization of the map? (not required if you just want to render a map, but required if you want more components?) 
- [x] consolidate ajax methods among components: Search component is using hand-rolled one, and routing component is using `corslite`. 
- [x] modularize `lrm-mapzen` so mapzenjs doesn't have to use global name to accept the instance.

### PR DESCRIPTION
**Pull request by hanbyul-here**
- closes #316 
- fetches `leaflet-routing-machine`, `lrm-mapzen` to embed in mapzen.js
- check for api check and options for components were redundant, extracted to api key manager file
- there is a commit saying `bumping up to 1.0.3`, but I took it back  since it makes more sense to be handled in different pr
- This is going to increase ~100kb for min, ~200kb for not-minified one

### CODE REVIEW

**Comment by hanbyul-here** [comment]
@rmglennon & @kkowalsky  : Syntax to use routing component will be similar, but there are more default option that we are offering through mapzen.js. Ex. Users don't have to pass `formatter` , `routeLine` (Unless users wants to use their own things) since mapzen.js sets default for them.

File: package.json [hunk 1]
```diff
@@ -20,9 +20,14 @@
     "leaflet": "global:L"
   },
   "dependencies": {
+    "corslite": "0.0.7",
```
Comment (hanbyul-here) [inline package.json]: I added these two lines just to pass CircleCI test pass. I tried to solve this, but then found out it is a bigger problem so should be handled separately. I will open another pr about this.

**Comment by hanbyul-here** [comment]
I will resolve the conflict after this pr is merged : https://github.com/mapzen/mapzen.js/pull/391

**Comment by hanbyul-here** [comment]
@rfriberg I will! thanks for the comment!

**Comment by kkowalsky** [comment]
:+1: @hanbyul-here

**Pull request merged**